### PR TITLE
Feature: Regular Grid Sampler

### PR DESCRIPTION
**Pull request by JakobEliasWagner**
# Feature: Regular Grid Sampler

## Description

This pull request introduces a new `RegularGridSampler` class, enhancing the sampling capabilities within n-dimensional hyperrectangles. The sampler is designed to generate samples that are as uniformly distributed across the grid as possible, with added functionality to handle over and undersampling when the requested number of samples cannot be perfectly achieved.

### Which issue does this PR tackle?

  - Regular grids currently need to be sampled manually.
  - For more general descriptions of functions, function-spaces and their discrete representations a multitude of different samplers and sampling strategies are required.

### How does it solve the problem?

  - Uniform Sampling: Efficiently samples from n-dimensional hyperrectangles to ensure a uniform distribution.
  - Flexible Sampling Options: Supports both oversampling and undersampling to facilitate sampling if the exact number of requested samples can not be achieved.

### How are the changes tested?

  - Added unit-tests to ensure functionality.

## Checklist for Contributors

- [x] Scope: This PR tackles exactly one problem.
- [x] Conventions: The branch follows the `feature/title-slug` convention.
- [x] Conventions: The PR title follows the `Bugfix: Title` convention.
- [x] Coding style: The code passes all pre-commit hooks.
- [x] Documentation: All changes are well-documented.
- [x] Tests: New features are tested and all tests pass successfully.
- [x] Changelog: Updated CHANGELOG.md for new features or breaking changes.
- [x] Review: A suitable reviewer has been assigned.

## Checklist for Reviewers:

- [x] The PR solves the issue it claims to solve and only this one.
- [x] Changes are tested sufficiently and all tests pass.
- [x] Documentation is complete and well-written.
- [x] Changelog has been updated, if necessary.

### CODE REVIEW

**Comment by JakobEliasWagner** [comment]
Currently the target branch is still `feature/sampler` as this feature is dependent on the features of this branch.

File: src/continuity/discrete/regular_grid.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        samples_per_dim = self.x_aspect * samples_per_dim
+        samples_per_dim = torch.max(
+            samples_per_dim, torch.ones(samples_per_dim.shape)
+        )  # ensure every dimension is sampled
+        return samples_per_dim
+
+    def adjust_samples_to_fit(
+        self, n_samples: int, samples_per_dim: torch.Tensor
+    ) -> torch.Tensor:
+        """Adjust the samples distribution to better fit the total_samples requirement.
```
Comment (samuelburbulla) [inline src/continuity/discrete/regular_grid.py:89]: Looks good, too, but please explain in the docstring.

**Review by samuelburbulla — CHANGES_REQUESTED** [review]
Looks good! Just make the rule how the grid is computed more concise in the class' docstring (preferably in formulas).

**Comment by samuelburbulla** [comment]
Please update the CHANGELOG as `RegularGridSampler` is a new feature.

File: src/continuity/discrete/regular_grid.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        >>> print(samples.shape)
+        torch.Size([125, 3])
+    """
+
+    def __init__(
+        self, x_min: torch.Tensor, x_max: torch.Tensor, prefer_more_samples: bool = True
+    ):
+        super().__init__(x_min, x_max)
+        self.prefer_more_samples = prefer_more_samples
+        self.x_aspect = self.x_delta / torch.sum(self.x_delta)
```
[the remark this answers is not in the document]
Comment (JakobEliasWagner) [reply src/continuity/discrete/regular_grid.py:44]: Added a test and changed the implementation to account for that.

**Pull request merged**